src/Forms/CommentForm.php

```diff
@@ -236,7 +236,29 @@ public function doPostComment($data, $form)
                 break;
         }
 
-        $comment = Comment::create();
+        // Updating comment
+        $existingCommentId = $data["CommentId"] ?? null;
+
+        if ($existingCommentId) {
+            $comment = Comment::get()->byID($existingCommentId);
+
+            if (!$comment) {
+                return $this->getRequestHandler()->httpError(404);
+            }
+
+            if (!$comment->canUpdate()) {
+                return Security::permissionFailure(
+                    $this->controller,
+                    _t(
+                        'SilverStripe\\Comments\\Controllers\\CommentingController.PERMISSIONFAILURE',
+                        "You're not able to update this comment. Please ensure you are logged in and have an "
+                        . 'appropriate permission level.'
```

Comment:
I'm not sure if the translation key should be the same as in the post comment failure since it's a different message. Perhaps we need a new key `PERMISSIONFAILURE_UPDATE' or similar.